plugins/woocommerce/src/Internal/EmailEditor/WCTransactionalEmails/WCEmailTemplateGenerator.php

```diff
@@ -14,60 +14,145 @@ class WCEmailTemplateGenerator {
 	 * @var WCEmailTemplateManager
 	 */
 	private $template_manager;
+
+	/**
+	 * Default templates.
+	 * @var array
+	 */
 	private $default_templates = [];
 
 	public function __construct() {
 		$this->template_manager = WCEmailTemplateManager::get_instance();
 	}
 
+	/**
+	 * Initialize the email template generator.
+	 *
+	 * This function initializes the email template generator by loading the default templates and generating initial email templates if needed.
+	 */
 	public function init() {
 		$this->init_default_templates();
 		$this->generate_initial_email_templates();
 	}
 
+	/**
+	 * Initialize the default templates.
+	 *
+	 * This function initializes the default templates for the core transactional emails.
+	 * It fetches all the emails from WooCommerce and filters them to include only the core transactional emails.
+	 *
+	 */
 	private function init_default_templates() {
-		$this->default_templates = [
-			'new_order' => [
-				'title' => __('New Order', 'woocommerce'),
-				'content' => $this->get_new_order_template(),
-				'enabled' => true
-			],
-			'customer_processing_order' => [
-				'title' => __('Processing Order', 'woocommerce'),
-				'content' => $this->get_processing_order_template(),
-				'enabled' => true
-			],
-			'customer_note' => [
-				'title' => __('Customer Note', 'woocommerce'),
-				'content' => $this->get_customer_note_template(),
-				'enabled' => true
-			],
-			// other email types...
-		];
+		$core_transactional_emails = WCTransactionalEmails::$core_transactional_emails;
+
+		$wc_emails = \WC_Emails::instance();
+		$email_types = $wc_emails->get_emails();
+		$email_types = array_filter( $email_types, function( $email ) use ( $core_transactional_emails ) {
+			return in_array( $email->id, $core_transactional_emails );
+		} );
```

Comment:
Just an idea. I think we could also support 3rd party emails by having universal email content that would contain only the woo_placeholder block. So when generating the email posts, we could check if there is a post content for the type, and if not, we would use the universal one. We could then allow 3rd party integrators to provide their own block content via a filter.
For now, I think core emails are fine, but this way, we can add basic support for correctly registered 3rd party emails soon.

Reply:
I share the same thoughts here. I would love to add a filter here to support creating initial email posts for third-party extensions, but I recall Pavel asking us to hold off on doing that for now. I may add it, but mark it as experimental.